slurm_train.sh

```diff
@@ -1,37 +1,49 @@
 #!/bin/bash -l
 #SBATCH --job-name=NeurWP
 #SBATCH --account=s83
-#SBATCH --partition=normal
+#SBATCH --time=24:00:00
 #SBATCH --nodes=1
-#SBATCH --ntasks-per-node=4
-#SBATCH --mem=400G
+#SBATCH --gres=gpu:8
+#SBATCH --ntasks-per-node=8
+#SBATCH --partition=normal
+#SBATCH --mem=375G
 #SBATCH --no-requeue
+#SBATCH --exclusive
 #SBATCH --output=lightning_logs/neurwp_out.log
 #SBATCH --error=lightning_logs/neurwp_err.log
 
-export PREPROCESS=false
-export NORMALIZE=false
+export PREPROCESS=true
+export NORMALIZE=true
+export DATASET="cosmo"
+export MODEL="hi_lam"
 
 # Load necessary modules
 conda activate neural-lam
 
 if [ "$PREPROCESS" = true ]; then
     echo "Create static features"
-    srun -ul -N1 -n1 python create_static_features.py --boundaries 60 --dataset "cosmo"
-    echo "Creating mesh"
-    srun -ul -N1 -n1 python create_mesh.py --dataset "cosmo" --plot 1
+    python create_static_features.py --boundaries 60 --dataset $DATASET
+    if [ "$MODEL" = "hi_lam" ]; then
+        echo "Creating hierarchical mesh"
+        python create_mesh.py --dataset $DATASET --plot 1 --graph hierarchical --levels 4 --hierarchical 1
+    else
+        echo "Creating flat mesh"
+        python create_mesh.py --dataset $DATASET --plot 1
+    fi
     echo "Creating grid features"
-    srun -ul -N1 -n1 python create_grid_features.py --dataset "cosmo"
+    python create_grid_features.py --dataset $DATASET
     if [ "$NORMALIZE" = true ]; then
         # This takes multiple hours!
         echo "Creating normalization weights"
-        srun -ul -N1 -n1 python create_parameter_weights.py --dataset "cosmo" --batch_size 32 --n_workers 8 --step_length 1
+        python create_parameter_weights.py --dataset $DATASET --batch_size 32 --n_workers 8 --step_length 1
     fi
 fi
 
 ulimit -c 0
-export OMP_NUM_THREADS=16
 
-# Run the script with torchrun
-srun -ul python train_model.py --dataset "cosmo" --val_interval 5 \
-    --epochs 10 --n_workers 6 --batch_size 8 --subset_ds 1
+echo "Training model"
+if [ "$MODEL" = "hi_lam" ]; then
```

Comment:
I think it is a bity ugly to not just be able to have the entire thing duplicated for an extra arg. But it would work

Reply:
yes. so I created these slurm scripts for developers first. and to switch between models, the checkpoint, the graph and the submission script must align. To make it easier for devs, I made it a simple switch at the top of the script that is hard to miss. I suggest to leave this for now for train, eval and predict. For MLOps (e.g. `slurm_offline.py` a more "operational" setup should be chosen.